### PR TITLE
Disabling generation of invalid atomic instructions

### PR DESCRIPTION
**Pull request by vmaksimo**
Some atomic instructions allow only integer result type

### CODE REVIEW

File: lib/SPIRV/OCL20ToSPIRV.cpp [hunk 1]
```diff
@@ -399,6 +399,12 @@ void OCL20ToSPIRV::visitCallInst(CallInst &CI) {
   }
   if (DemangledName.find(kOCLBuiltinName::AtomicPrefix) == 0 ||
       DemangledName.find(kOCLBuiltinName::AtomPrefix) == 0) {
+
+    // Compute atomic bulitins do not support floating types.
```
Comment (Hopetech) [inline lib/SPIRV/OCL20ToSPIRV.cpp]: Small nit:
s/bulitins/builtins/g

**Comment by Hopetech** [comment]
Should we need to add the same security in OCL21ToSPIRV too?

File: lib/SPIRV/OCLUtil.cpp [hunk 2]
```diff
@@ -107,6 +107,32 @@ size_t getAtomicBuiltinNumMemoryOrderArgs(StringRef Name) {
...
 }
 
+bool isComputeAtomicOCLBuiltin(StringRef DemangledName) {
+  if (!DemangledName.startswith(kOCLBuiltinName::AtomicPrefix) &&
+      !DemangledName.startswith(kOCLBuiltinName::AtomPrefix))
+    return false;
+
+  return llvm::StringSwitch<bool>(DemangledName)
+      .EndsWith("add", true)
+      .EndsWith("sub", true)
```
Comment (Hopetech) [inline lib/SPIRV/OCLUtil.cpp]: It seems that we missed
`.EndsWith("cmpxchg", true)`

**Comment by Hopetech** [comment]
> > Should we need to add the same security in OCL21ToSPIRV too?
> 
> I don't so. OCL21ToSPIRV processes OpenCL 21 specific built-in functions and AFAIK OpenCL 21 does not add new atomic built-in functions, so it should enough to handle the builtins processed in OCL20ToSPIRV.

Ack. Sorry for the noise.

File: lib/SPIRV/OCLUtil.cpp (hunk 2 above)
Comment (AlexeySotkin) [reply lib/SPIRV/OCLUtil.cpp]: xchg allows other then int result types. i.e. float. 
Also I think xchg doesn't do any computation, just access memory. So we don't need it here.

**Pull request merged**